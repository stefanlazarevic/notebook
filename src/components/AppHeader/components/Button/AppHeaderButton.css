.AppHeaderButton {
  /* Reset */
  appearance: none;
  border: 0;
  cursor: pointer;
  outline: 0;

  /* Dimensions */
  width: 40px;
  height: 40px;

  /* Content */
  font-size: 24px; /* @todo Switch to REM. */
  line-height: 1.4rem;

  /* Colors */
  background-color: var(--app-background-5-lighter-color);
  color: var(--app-text-base-color);

  /* Border */
  border-radius: 4px;
}

.AppHeaderButton:hover,
.AppHeaderButton:focus {
  /* Colors */
  background-color: var(--app-background-10-lighter-color);
  color: var(--app-text-5-lighter-color);
}

.AppHeaderButton:disabled {
  /* Colors */
  color: var(--app-text-40-darker-color);
}
