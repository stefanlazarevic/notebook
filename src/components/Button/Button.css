.Button {
  box-sizing: border-box;

  position: relative;

  cursor: pointer;
}
