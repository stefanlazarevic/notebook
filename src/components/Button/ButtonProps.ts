import PropTypes, { InferProps, Validator } from "prop-types";

export const ButtonPropTypes = {
	/**
	 *
	 */
	id: PropTypes.string as Validator<string | undefined>,
	/**
	 *
	 */
	testid: PropTypes.string as Validator<string | undefined>,
	/**
	 *
	 */
	className: PropTypes.string as Validator<string | undefined>,
	/**
	 *
	 */
	children: PropTypes.node as Validator<React.ReactNode | React.ReactNodeArray | undefined>,
	/**
	 *
	 */
	title: PropTypes.string as Validator<string | undefined>,
	/**
	 *
	 */
	lang: PropTypes.string as Validator<string | undefined>,
	/**
	 *
	 */
	disabled: PropTypes.bool as Validator<boolean | undefined>,
	/**
	 *
	 */
	onClick: PropTypes.func as Validator<
		(event: React.SyntheticEvent<HTMLButtonElement>) => void | undefined
	>,
	/**
	 *
	 */
	"aria-label": PropTypes.string as Validator<string | undefined>,
	/**
	 *
	 */
	"aria-labelledby": PropTypes.string as Validator<string | undefined>,
	/**
	 *
	 */
	"aria-pressed": PropTypes.oneOf([true, false, "true", "false", "mixed"]) as Validator<
		boolean | "false" | "mixed" | "true" | undefined
	>,
	/**
	 *
	 */
	"aria-expanded": PropTypes.oneOf([true, false, "true", "false"]) as Validator<
		boolean | "false" | "true" | undefined
	>,
	/**
	 *
	 */
	"aria-haspopup": PropTypes.oneOf([true, false, "true", "false", "menu"]) as Validator<
		boolean | "true" | "false" | "menu" | undefined
	>,
};

export type ButtonProps = InferProps<typeof ButtonPropTypes>;
