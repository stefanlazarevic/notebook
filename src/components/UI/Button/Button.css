.Button {
  appearance: none;

  position: relative;

  border: 0;

  outline: 0;

  cursor: pointer;

  background: transparent;

  padding: 0;
}
