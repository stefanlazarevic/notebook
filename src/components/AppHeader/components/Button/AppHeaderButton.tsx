import React from "react";

import "./AppHeaderButton.css";

import { AppHeaderButtonProps } from "./AppHeaderButtonProps";

export default function AppHeaderButton(props: AppHeaderButtonProps) {
  return (
    <button
      className="AppHeaderButton"
      disabled={props.disabled}
      title={props.title}
    >
      {props.children}
    </button>
  );
}
