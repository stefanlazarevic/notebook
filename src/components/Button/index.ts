import Button from "./Button";
import BackButton from "./components/BackButton";
import CloseButton from "./components/CloseButton";
import DownloadButton from "./components/DownloadButton";
import EyeButton from "./components/EyeButton";
import HelpButton from "./components/HelpButton";
import NextButton from "./components/NextButton";
import PreviousButton from "./components/PreviousButton";

export { BackButton, CloseButton, DownloadButton, EyeButton, HelpButton, NextButton, PreviousButton };

export default Button;
