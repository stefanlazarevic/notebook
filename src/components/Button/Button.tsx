import React, { useCallback } from "react";

import "./Button.css";

import { ButtonProps, ButtonPropTypes } from "./ButtonProps";

import useClassNames from "../Utils/hooks/classNames";

function Button(props: ButtonProps) {
	const className = useClassNames("Button", props.className);

	const onKeyDown = useCallback(
		(event: React.KeyboardEvent<HTMLButtonElement>) => {
			if (typeof props.onClick === "function") {
				props.onClick(event);
			}
		},
		[props.onClick]
	);

	return (
		<button
			id={props.id}
			data-testid={props.testid}
			className={className}
			title={props.title}
			lang={props.lang}
			disabled={props.disabled}
			aria-label={props["aria-label"]}
			aria-labelledby={props["aria-labelledby"]}
			aria-pressed={props["aria-pressed"]}
			aria-expanded={props["aria-expanded"]}
			aria-haspopup={props["aria-haspopup"]}
			onClick={props.onClick}
			onKeyDown={onKeyDown}
		>
			{props.children}
		</button>
	);
}

Button.propTypes = ButtonPropTypes;

Button.defaultProps = {
	onClick: undefined,
};

Button.displayName = "Button";

export default Button;
