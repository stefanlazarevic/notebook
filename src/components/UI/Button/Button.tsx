import React from "react";

import "./Button.css";

import { ButtonProps } from "./ButtonProps";

export default function Button(props: ButtonProps) {
  const classnames = ["Button", props.className];

  return (
    <button {...props} className={classnames.join("")}>
      {props.children}
    </button>
  );
}
