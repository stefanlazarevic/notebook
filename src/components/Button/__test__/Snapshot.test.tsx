import React from "react";
import { render, cleanup } from "@testing-library/react";
import Button from "../index";

afterEach(cleanup);

it("Подразумеванa својства се подударају са последњим снимком.", () => {
	const { asFragment } = render(<Button />);

	expect(asFragment()).toMatchSnapshot();
});

it("Сва подешена својства се подударају са последњим снимком.", () => {
	const { asFragment } = render(
		<Button
			id="id"
			testid="test-id"
			className="TestButton"
			title="Button title"
			lang="en"
			disabled={false}
			aria-label="test-label-text"
			aria-labelledby="test-label"
			aria-pressed={"mixed"}
			aria-expanded={false}
			aria-haspopup={false}
		>
			Test
		</Button>
	);

	expect(asFragment()).toMatchSnapshot();
});
