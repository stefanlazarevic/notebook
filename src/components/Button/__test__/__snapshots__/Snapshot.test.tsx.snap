// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Подразумеванa својства се подударају са последњим снимком. 1`] = `
<DocumentFragment>
  <button
    class="Button"
  />
</DocumentFragment>
`;

exports[`Сва подешена својства се подударају са последњим снимком. 1`] = `
<DocumentFragment>
  <button
    aria-expanded="false"
    aria-haspopup="false"
    aria-label="test-label-text"
    aria-labelledby="test-label"
    aria-pressed="mixed"
    class="Button TestButton"
    data-testid="test-id"
    id="id"
    lang="en"
    title="Button title"
  >
    Test
  </button>
</DocumentFragment>
`;
